Code/Editor/GameExporter.cpp

```diff
@@ -136,7 +136,7 @@ bool CGameExporter::Export(unsigned int flags, [[maybe_unused]] EEndian eExportE
             m_settings.SetHiQuality();
         }
 
-        CryAutoLock<CryMutex> autoLock(CGameEngine::GetPakModifyMutex());
+        AZStd::lock_guard<AZStd::recursive_mutex> autoLock(CGameEngine::GetPakModifyMutex());
```

Comment:
Prefer [scoped_lock](https://en.cppreference.com/w/cpp/thread/scoped_lock) to [lock_guard](https://en.cppreference.com/w/cpp/thread/lock_guard#Notes).

Also [class template argument deduction](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction) also known as CTAD is able to determine the template parameter based on the constructor for AZStd::scoped_lock or AZStd::lock_guard

```suggestion
        AZStd::scoped_lock autoLock(CGameEngine::GetPakModifyMutex());
```